meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh

```diff
@@ -731,7 +721,217 @@ verify_artifacts() {
+
+  grep -q "Gigabit Ethernet region" <<<  "$FLASH_REGIONS" && BOARD_HAS_GBE_REGION=1
+  grep -qE "Gigabit Ethernet region.*read-write" <<<  "$FLASH_REGIONS" && BOARD_GBE_REGION_RW=1
+  grep -qE "Gigabit Ethernet region.*locked" <<< "$FLASH_REGIONS" && BOARD_GBE_REGION_LOCKED=1
+}
+
+check_blobs_in_binary() {
+  BINARY_HAS_FD=0
+  BINARY_HAS_ME=0
+
+  # If there is no descriptor, there is no ME as well, so skip the check
+  if [ $BOARD_HAS_FD_REGION -ne 0 ]; then
+    ME_OFFSET=$(ifdtool -d $1 2> /dev/null | grep "Flash Region 2 (Intel ME):" | sed 's/Flash Region 2 (Intel ME)\://' |awk '{print $1;}')
+    # Check for IFD signature at offset 0 (old descriptors)
+    if [ $(tail -c +0 $1|head -c 4|xxd -ps) == "5aa5f00f" ]; then
+      BINARY_HAS_FD=1
+    fi
+    # Check for IFD signature at offset 16 (new descriptors)
+    if [ $(tail -c +17 $1|head -c 4|xxd -ps) == "5aa5f00f" ]; then
+      BINARY_HAS_FD=1
+    fi
+    # Check for ME FPT signature at ME offset + 16 (old ME)
+    if [ $(tail -c +$((0x$ME_OFFSET + 17)) $1|head -c 4|tr -d '\0') == "\$FPT" ]; then
+      BINARY_HAS_ME=1
+    fi
+    # Check for aa55 signature at ME offset + 4096 (new ME)
+    if [ $(tail -c +$((0x$ME_OFFSET + 4097)) $1|head -c 2|xxd -ps) == "aa55" ]; then
+      BINARY_HAS_ME=1
+    fi
+  fi
+}
+
+check_if_me_disabled() {
+
+  ME_DISABLED=0
+
+  if [ $BOARD_HAS_ME_REGION -eq 0 ]; then
+    # No ME region
+    ME_DISABLED=1
+    return
+  fi
+
+  # Check if HECI present
+  # FIXME: what if HECI is not device 16.0?
+  if [ -d /sys/class/pci_bus/0000:00/device/0000:00:16.0 ]; then
+    # Check ME Current Operation Mode at offset 0x40 bits 19:16
+    ME_OPMODE="$(setpci -s 00:16.0 42.B 2> /dev/null | cut -c2-)"
+    if [ $ME_OPMODE == "0" ]; then
+      echo "ME is not disabled"  >> $ERR_LOG_FILE
+      return
+    elif [ $ME_OPMODE == "2" ]; then
+      echo "ME is disabled (HAP/Debug Mode)"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "3" ]; then
+      echo "ME is soft disabled (HECI)"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "4" ]; then
+      echo "ME disabled by Security Override Jumper/FDOPS"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "5" ]; then
+      echo "ME disabled by Security Override MEI Message/HMRFPO"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "5" ]; then
+      echo "ME disabled by Security Override MEI Message/HMRFPO"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "7" ]; then
+      echo "ME disabled (Enhanced Debug Mode) or runs Ignition FW"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    else
+      print_warning "Unknown ME operation mode, assuming enabled."
+      echo "Unknown ME operation mode, assuming enabled."  >> $ERR_LOG_FILE
+      return
+    fi
+  else
+    # If we are running coreboot, check for status in logs
```

Comment:
Even if we are not running coreboot we should be able to check ME FWSTS from SMBIOS. Intel reference code does expose FWSTS in SMBIOS and this could be the method to check ME status. I did the exact same thing here: https://github.com/Dasharo/coreboot/pull/453 (has to be done per board right now, but maybe I will get a better idea how to do it more generic way).

Reply:
we can read FWSTS via /sys/class/mei/fw_status (needs mei_me driver probably)